src/Language/DifferentialDatalog/Attribute.hs

```diff
@@ -157,6 +164,23 @@ tdefGetCustomSerdeAttr d tdef =
          Left e  -> error e
          Right b -> b
 
+{- 'custom_from_record' attribute: Tells DDlog not to generate `FromRecord`
+ - implementation for a type.  The user must write their own implementations in
```

Comment:
it would be nice if the signature of this implementation was documented someplace.

Reply:
I don't think users should be writing these annotations. This is an ugly workaround for the OVSDB problem that I could not solve properly without a major type system overhaul.